os/fs/mount/fs_mount.c

```diff
@@ -331,17 +332,42 @@ int mount(FAR const char *source, FAR const char *target, FAR const char *filesy
 		 * error.
 		 */
 
-		fdbg("ERROR: Bind method failed: %d\n", ret);
+		if (ret == -ENODEV && strncmp(filesystemtype, "smartfs", 7) == 0) {
+			fdbg("Reformatting the SMART devce and retrying to mount\n");
+#ifndef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+			ret = mksmartfs(source, true);
+#else
+			ret = mksmartfs(source, 1, true);
```

Comment:
This is very dangerous actually, during power cut time, we can't read sector 0 properly even if 0 is exist properly, because of low electronic power.
So we allowed format at the start of factory process or by user command.

Reply:
We have a problem that if not mounted during boot time, the partition can never be mounted with a SmartFs instance.

But I think you mean to say here that if we face power cut during manual mounting from TASH
We cannot guarantee proper format of smartfs mtd partition after reboot?